ReactAndroid/src/main/java/com/facebook/react/modules/settings/SettingsModule.java

```diff
@@ -0,0 +1,180 @@
+  static final String TAG = "SettingsMmodule";
+
+  private static boolean sSettingsModuleInstantiated = false;
+
+  static private String sFilename = "";
+  private Boolean ignoringUpdates = false;
+  private SharedPreferences mPreferences;
+
+  public SettingsModule(ReactApplicationContext context) {
+    super(context);
+
+  }
+
+  @Override
+  public String getName() {
+    return SettingsModule.NAME;
+  }
+
+  private SharedPreferences ensurePreferences() {
+    if (mPreferences == null) {
+      if (SettingsModule.sFilename.length() > 0) {
+        mPreferences = getReactApplicationContext().getSharedPreferences(SettingsModule.sFilename, Context.MODE_PRIVATE);
+      } else if (getCurrentActivity() != null) {
+        mPreferences = getCurrentActivity().getPreferences(Context.MODE_PRIVATE);
+      } else {
+        mPreferences = getReactApplicationContext().getSharedPreferences(DEFAULT_FILE_NAME, Context.MODE_PRIVATE);
+      }
+      sSettingsModuleInstantiated = true;
+      mPreferences.registerOnSharedPreferenceChangeListener(this);
+    }
+
+    return mPreferences;
+  }
+
+  /***
+   * Configure the Settings Module to use a SharedPreferences file with a specific name as its backing store.
+   * Useful if the Settings Module needs to use an existing SharedPreferences file or to ensure that Settings Modules
+   * belonging to different ReactPackages will all share the same SharedPreferences file for their backing store.
+   * <p></p>
+   * <b>Note:</b>this method may only be called before a Settings Module initialization (i.e. before a ReactMainPackage is instantiated)
+   * An app would typically invoke this function in the app's main activity before setting up the React Native environment
+   * @param filename the name of the SharedPreferences file to use or create for the backing store
+   * @throws IllegalStateException if called after instantiation of a Settings module
+   */
+  static public void setSharedPreferencesFilename(String filename) {
+    if (sSettingsModuleInstantiated) {
+      throw new IllegalStateException("Attempt to set Settings Module SharedPreferences file name after a Settings Module instantiation");
+    }
+    sFilename = filename;
+  }
+
+  @Override
+  public void onCatalystInstanceDestroy() {
+    ensurePreferences().unregisterOnSharedPreferenceChangeListener(this);
+  }
+
+  @ReactMethod
+  public void setValues(ReadableMap values) {
+    this.ignoringUpdates = true;
+
+    ReadableMapKeySetIterator iterator = values.keySetIterator();
+    SharedPreferences.Editor editor = mPreferences.edit();
+
+    while (iterator.hasNextKey()) {
+      String key = iterator.nextKey();
+      ReadableType type = values.getType(key);
+
+      switch (type) {
+        case Null:
+          editor.remove(key);
+          break;
+        case Boolean:
+          editor.putBoolean(key, values.getBoolean(key));
+          break;
+        case Number:
+          editor.putFloat(key, (float) values.getDouble(key));
```

Comment:
Maybe the easiest and the best way to handle this would be to put a simple check to the set method in Javascript. It would make sure only non-null strings and numbers would be saved.